wormhole-connect/src/hooks/useSortedRoutesWithQuotes.ts

```diff
@@ -0,0 +1,118 @@
+import { useMemo } from 'react';
+import { useSelector } from 'react-redux';
+import type { RootState } from 'store';
+import { routes, amount as sdkAmount } from '@wormhole-foundation/sdk';
+import useRoutesQuotesBulk from 'hooks/useRoutesQuotesBulk';
+import config from 'config';
+import { RouteState } from 'store/transferInput';
+
+type Quote = routes.Quote<
+  routes.Options,
+  routes.ValidatedTransferParams<routes.Options>
+>;
+type RouteWithQuote = {
+  route: RouteState;
+  quote: Quote;
+};
+
+type HookReturn = {
+  allSupportedRoutes: RouteState[];
+  sortedRoutes: RouteState[];
+  quotesMap: ReturnType<typeof useRoutesQuotesBulk>['quotesMap'];
+  isFetchingQuotes: boolean;
+};
+
+export const useSortedRoutesWithQuotes = (): HookReturn => {
+  const { amount, routeStates, fromChain, token, toChain, destToken } =
+    useSelector((state: RootState) => state.transferInput);
+  const { toNativeToken } = useSelector((state: RootState) => state.relay);
+
+  const supportedRoutes = useMemo(
+    () => (routeStates || []).filter((rs) => rs.supported),
+    [routeStates],
+  );
+
+  const supportedRoutesNames = useMemo(
+    () => supportedRoutes.map((r) => r.name),
+    [supportedRoutes],
+  );
+
+  const { quotesMap, isFetching } = useRoutesQuotesBulk(supportedRoutesNames, {
+    amount,
+    sourceChain: fromChain,
+    sourceToken: token,
+    destChain: toChain,
+    destToken,
+    nativeGas: toNativeToken,
+  });
+
+  const routesWithQuotes = useMemo(() => {
+    return supportedRoutes
+      .map((route) => {
+        const quote = quotesMap[route.name];
+        if (quote?.success) {
+          return {
+            route,
+            quote,
+          };
+        } else {
+          return undefined;
+        }
+      })
+      .filter(Boolean) as RouteWithQuote[];
```

Comment:
we should still show routes that received a `MinAmountError` - so the user can see that there are other routes available if they increase their size